typescript/sdk/src/ism/EvmIsmModule.hardhat-test.ts

```diff
@@ -0,0 +1,720 @@
+    await hre.ethers.provider.send('hardhat_impersonateAccount', [account]);
+    await fundingAccount.sendTransaction({
+      to: account,
+      value: hre.ethers.utils.parseEther('1.0'),
+    });
+    return MultiProvider.createTestMultiProvider({
+      signer: hre.ethers.provider.getSigner(account),
+    });
+  }
+
+  async function logAndSendTransactions(txs: AnnotatedEthersV5Transaction[]) {
+    for (const tx of txs) {
+      if (tx.annotation) {
+        console.log(tx.annotation);
+      }
+      await multiProvider.sendTransaction(chain, tx.transaction);
+    }
+  }
+
+  describe('create', async () => {
+    it('deploys a simple ism', async () => {
+      const config = randomMultisigIsmConfig(3, 5);
+
+      const ism = await createIsmModule({ config });
+      const matches = await ismModuleMatchesConfig({
+        ism,
+        config,
+        configured: false,
+      });
+      expect(matches).to.be.true;
+    });
+
+    it('deploys a trusted relayer ism', async () => {
+      const relayer = randomAddress();
+      const config: TrustedRelayerIsmConfig = {
+        type: IsmType.TRUSTED_RELAYER,
+        relayer,
+      };
+
+      const ism = await createIsmModule({ config });
+      const matches = await ismModuleMatchesConfig({
+        ism,
+        config,
+        configured: false,
+      });
+      expect(matches).to.be.true;
+    });
+
+    for (const type of [IsmType.ROUTING, IsmType.FALLBACK_ROUTING]) {
+      it(`deploys ${type} routingIsm with correct routes`, async () => {
+        exampleRoutingConfig.type = type as
+          | IsmType.ROUTING
+          | IsmType.FALLBACK_ROUTING;
+
+        const ism = await createIsmModule({ config: exampleRoutingConfig });
+        const matches = await ismModuleMatchesConfig({
+          ism,
+          config: exampleRoutingConfig,
+          configured: false,
+        });
+        expect(matches).to.be.true;
+      });
+    }
+
+    for (let i = 0; i < 16; i++) {
+      it(`deploys a random ism config #${i}`, async () => {
+        const config = randomIsmConfig();
+        let ism: EvmIsmModule;
+        try {
+          ism = await createIsmModule({ config });
+        } catch (e) {
+          console.error('Failed to deploy random ism config', e);
+          console.error(stringifyObject(config as object, 'json', 2));
+          process.exit(1);
+        }
+
+        try {
+          const matches = await ismModuleMatchesConfig({
+            ism,
+            config,
+            configured: false,
+          });
+          expect(matches).to.be.true;
+        } catch (e) {
+          console.error('Failed to match random ism config', e);
+          console.error('EXPECTED');
+          console.error(stringifyObject(config as object, 'yaml', 2));
+          console.error('ACTUAL');
+          console.error(
+            stringifyObject((await ism.read()) as object, 'yaml', 2),
+          );
+          process.exit(1);
+        }
+      });
```

Comment:
the catch stuff seems useful for debugging these tests but not necessary in the published tests

Reply:
we didnt really discuss this but I expected these tests to basically be
```typescript
const ismConfig = randomIsmConfig();
const ismModule = await createIsmModule({config});
const derivedConfig = await ismModule.read();
expect(derivedConfig).to.deep.equal(ismConfig);
```

Reply:
relates back to these two
- https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3759/files#r1608999641
- https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3759/files#r1609006616